binary_transparency/firmware/internal/verify/bundle.go

```diff
@@ -34,53 +34,53 @@ type ConsistencyProofFunc func(from, to uint64) ([][]byte, error)
 // the one in the bundle. It also checks consistency proof between update log point
 // and device log point (for non zero device tree size). Upon successful verification
 // returns a proof bundle
-func BundleForUpdate(bundleRaw, fwHash []byte, dc api.LogCheckpoint, cpFunc ConsistencyProofFunc) (api.ProofBundle, error) {
+func BundleForUpdate(bundleRaw, fwHash []byte, dc api.LogCheckpoint, cpFunc ConsistencyProofFunc) (api.ProofBundle, api.FirmwareMetadata, error) {
 	proofBundle, fwMeta, err := verifyBundle(bundleRaw)
 	if err != nil {
-		return proofBundle, err
+		return proofBundle, fwMeta, err
 	}
 
 	if got, want := fwHash, fwMeta.FirmwareImageSHA512; !bytes.Equal(got, want) {
-		return proofBundle, fmt.Errorf("firmware update image hash does not match metadata (0x%x != 0x%x)", got, want)
+		return proofBundle, fwMeta, fmt.Errorf("firmware update image hash does not match metadata (0x%x != 0x%x)", got, want)
 	}
 
 	cProof, err := cpFunc(dc.TreeSize, proofBundle.Checkpoint.TreeSize)
 	if err != nil {
-		return proofBundle, fmt.Errorf("cpFunc failed: %q", err)
+		return proofBundle, fwMeta, fmt.Errorf("cpFunc failed: %q", err)
 	}
 
 	// Verify the consistency proof between device and bundle checkpoint
 	if dc.TreeSize > 0 {
 		lv := NewLogVerifier()
 		if err := lv.VerifyConsistencyProof(int64(dc.TreeSize), int64(proofBundle.Checkpoint.TreeSize), dc.RootHash, proofBundle.Checkpoint.RootHash, cProof); err != nil {
-			return proofBundle, fmt.Errorf("failed verification of consistency proof %w", err)
+			return proofBundle, fwMeta, fmt.Errorf("failed verification of consistency proof %w", err)
 		}
 	}
-	return proofBundle, nil
+	return proofBundle, fwMeta, nil
 }
 
-// BundleValidateWitness verifies the received bundle against witness checkpoint
-func BundleValidateWitness(pb api.ProofBundle, wc api.LogCheckpoint, cpFunc ConsistencyProofFunc) error {
+// BundleValidateRemote verifies the received bundle against a remotely retrieved checkpoint (e.g. witness).
+func BundleValidateRemote(pb api.ProofBundle, rc api.LogCheckpoint, cpFunc ConsistencyProofFunc) error {
```

Comment:
The naming in here was intended to be read with the package name (e.g. "verify Bundle For Update"), how about renaming this method to `verify.Consistency` and just pass in `pb.Checkpoint` and `rc`?

Reply:
renamed to `verify.BundleConsistency` as it verifies the bundle is consistent. It still needs a little other spice than the bundle and a checkpoint as its otherwise stateless and needs access to a client. That's what the `cpFunc` provides.